src/js/apps/forms/form/form_app.js

```diff
@@ -152,9 +152,21 @@ export default App.extend({
     this.responses.unshift(response);
     this.setState({ responseId: response.id });
   },
-  onFormServiceError() {
+  onFormServiceError(errors) {
     if (this.loadingModal) this.loadingModal.destroy();
 
+    if (errors[0].status === 403) {
+      this.isLocked = true;
+
+      this.showLockedSubmit();
```

Comment:
I'm on the fence about this:

really I think the only thing we need to do here is show the error and let it go back to save.  If we lock it, the user can't do anything to resolve the issue.  if their assignment gets flipped, even by them, they could potentially flip it back and submit.

Plus the other issue is that plenty of our forms, the major ones, have submit buttons within the form themselves..

Reply:
That's a good point. Didn't think of that. I agree the submit should stay, without doing the locking.